bots/elasticsearch/bot.py

```diff
@@ -25,98 +25,204 @@ def chunk(iterable, size):
-                            # If you use UTF-8 text with many non-ASCII characters, you may want to set the limit to 32766 / 3 = 10922 since UTF-8 characters may occupy at most 3 bytes
-                            'raw': {'type': 'string', 'index': 'not_analyzed', 'ignore_above': 10922}
-                        }
-                    }
-                }
-            }, {
-                'exact_matches': {
-                    'match': 'id',
-                    'mapping': {
-                        'enabled': False
-                    }
-                }
-            }],
+            'dynamic': False,
             'properties': {
-                'title': {
+                'affiliations': {
                     'type': 'string',
-                    'fields': {
-                        # From Elasticsearch documentation:
-                        # The value for ignore_above is the character count, but Lucene counts bytes.
-                        # If you use UTF-8 text with many non-ASCII characters, you may want to set the limit to 32766 / 3 = 10922 since UTF-8 characters may occupy at most 3 bytes
-                        'raw': {'type': 'string', 'index': 'not_analyzed', 'ignore_above': 10922}
-                    }
+                    'fields': EXACT_FIELD
                 },
-                'tags': {'type': 'string', 'index': 'not_analyzed'},
-                'subjects': {'type': 'string', 'index': 'not_analyzed'},
-                'type': {
+                'contributors': {
                     'type': 'string',
-                    'fields': {
-                        'raw': {'type': 'string', 'index': 'not_analyzed'}
-                    }
+                    'fields': EXACT_FIELD
                 },
-                'types': {
+                'date': {
+                    'type': 'date',
+                    'format': 'strict_date_optional_time',
+                    'include_in_all': False,
+                },
+                'date_created': {
+                    'type': 'date',
+                    'format': 'strict_date_optional_time',
+                    'include_in_all': False,
+                },
+                'date_modified': {
+                    'type': 'date',
+                    'format': 'strict_date_optional_time',
+                    'include_in_all': False,
+                },
+                'date_published': {
+                    'type': 'date',
+                    'format': 'strict_date_optional_time',
+                    'include_in_all': False,
+                },
+                'date_updated': {
+                    'type': 'date',
+                    'format': 'strict_date_optional_time',
+                    'include_in_all': False,
+                },
+                'description': {
                     'type': 'string',
                     'fields': {
-                        'raw': {'type': 'string', 'index': 'not_analyzed'}
+                        'en': {
```

Comment:
Oh, I should probably just delete the `en` fields. They are included in `_all`, but the values are copied before analysis, so there's no actual benefit unless we explicitly include `title.en`/`description.en` in the query. Exposing `query_string` syntax to the user makes it harder to do fun things with analyzers, unfortunately.